network/internal/testutils/testUtil.go

```diff
@@ -126,121 +114,107 @@ func NewTagWatchingConnManager(log zerolog.Logger, metrics module.LibP2PConnecti
 	}, nil
 }
 
-// GenerateIDs is a test helper that generate flow identities with a valid port and libp2p nodes.
-func GenerateIDs(t *testing.T, logger zerolog.Logger, n int, opts ...func(*optsConfig)) (flow.IdentityList,
-	[]p2p.LibP2PNode,
-	[]observable.Observable) {
-	libP2PNodes := make([]p2p.LibP2PNode, n)
-	tagObservables := make([]observable.Observable, n)
+// LibP2PNodeFixture is a test helper that generate flow identities with a valid port and libp2p nodes.
+// Note that the LibP2PNode created by this fixture is meant to used with a network and middleware components.
+// If you want to create a LibP2PNode for testing purpose, please use p2ptest.NodeFixture.
```

Comment:
Curious why we can't combine `p2ptest.NodeFixture` and `LibP2PNodeFixture` so that the same fixture can be used in both cases (stand alone p2p node and for use with middleware and network).

Reply:
We eventually plan to consolidate them, but for now, the notable blocker is the [handler function of libp2p node](https://github.com/onflow/flow-go/blob/a572dec7e140a59cf61ec7fe0b0466d2a43248f6/network/p2p/test/fixtures.go#L74), which is the call-back for delivery of received messages. When it is used stand-alone, the handler function is set to no-op. However, when the node is configured within the networking stack, the handler function is set by middleware. As this is a dependency injection pattern, we don't allow a handler function to be set if one is already configured. We will raise a panic to ensure the dependency injection pattern is not violated. Hence, the test fixture has only a single chance of setting the handler function without overriding it. We are still attempting to find a clear solution for this, especially with the long-term goal of removing the middleware; this may not be acting as a blocker anymore.